### DISCUSSION TITLE
Stat Block Parser Format?

### DISCUSSION THREAD

**Discussion by anoriginalmoniker** [deltagreen-foundryvtt/delta-green-foundry-vtt-system#168] [General]
Hi there. Thanks for all of your work on this. Loving everything so far.

I keep most of my notes and prep in markdown in Obsidian, and I'm working on creating a little export of my NPCs, so I could write up my characters in my note-taking app, then paste them into your parser. I've manage to follow the format from the official Arc Dream material, and it works great...except for the weapons part. Can you tell me how to best list the information for weapons to achieve the most accuracy. Here's an example of my text and a screenshot of what it produces. The percentage is the relevant skill, in this case firearms.

Thanks again!

Assault Riffle 17%, damage 1d12.
Submachine Gun 17%, damage Lethality 10%.

![Foundry_Virtual_Tabletop_IxDsUZovmW](https://github.com/user-attachments/assets/c8189336-580a-444b-b5a7-38aaee0426d8)

**Comment by TheLastScrub** [comment]
Sorry I meant to reply to this, and then forgot about it. One thing to note, I don't currently try to grab the skill (e.g. 'Firearms') and link it with the stat block parser, I think I set everything to 'custom' and give it a fixed % value.

The attacks section varies a lot so I have a lot of trouble grabbing this information... But I based a lot of my parsing work on the stat blocks in Impossible Landscapes, so something like this is going to be your best bet:

(A certain NPC)
(information about that NPC)
STR 14 CON 12 DEX 11 INT 15 POW 10 CHA 16
HP 13 WP 10 SAN 0
MOTIVATIONS AND DISORDERS: To do a spoilerish thing
To do another spoilerish thing
To do a third spoilerish thing
SKILLS: Alertness 40%, Anthropology 30%, Art (History)
30%, Bureaucracy 30%, Craft (Mechanics) 20%,
Dodge 30%, Drive 30%, Firearms 40%, Foreign
Language (Arabic) 65%, Foreign Language (Berber)
45%, Foreign Language (English) 70%, Foreign
Language (French) 50%, Foreign Language (Spanish)
20%, Foreign Language (Tartessian) 15%, History 70%,
HUMINT 70%, Persuade 80%, Ride 30%, Stealth 30%.
ATTACKS: Browning FN Model 1903 pistol 40%,damage 1D10.
Unarmed 40%, damage 1D4.

Where it's important for the parser that a given attack ALL BE ON THE SAME LINE, so you may need to do a bit of editing of the stat blocks in the books if you use them directly.

If you have a certain stat block you are trying to use that seems like it should work, you could drop it in here, and I could see if I can determine why it isn't working.
